frame/staking/src/lib.rs

```diff
@@ -655,11 +679,14 @@ impl Default for Forcing {
 	fn default() -> Self { Forcing::NotForcing }
 }
 
+const HISTORY_DEPTH: EraIndex = 84;
```

Comment:
Why is this not configurable in the Trait or as a storage item?

Reply:
we can make it configurable in the Trait for no cost,
we can also make it configurable and changeable dynamically:
* either by adding a storage for previous removed era
* or by requiring some migration code when this value is changed

Reply:
You understand the code the best now. Do you suspect that governance will want to update this value? i.e. make a vote to increase or decrease the history depth?

If so, it should be a storage value.

Also depends on how often the value is accessed.

Reply:
It is only accessed once per new era so a storage item totally makes sense. It seems the current logic may not be perfectly sound if the value is updated dynamically, i.e.

```
		// Clean old era information.
		if let Some(old_era) = current_era.checked_sub(HISTORY_DEPTH) {
			<ErasStakers<T>>::remove_prefix(old_era);
			<ErasStakersClipped<T>>::remove_prefix(old_era);
			<ErasValidatorPrefs<T>>::remove_prefix(old_era);
			<ErasValidatorReward<T>>::remove(old_era);
			<ErasRewardPoints<T>>::remove(old_era);
			<ErasTotalStake<T>>::remove(old_era);
			ErasStartSessionIndex::remove(old_era);
		}
```

If governance reduced the era history, some state will be not cleaned up.

This could be addressed in the extrinsic function which updates the storage item and also cleans up any old state which may have been missed.